### PR TITLE
[DataFrame] Fixing the code formatting of the tests

### PR DESCRIPTION
**Pull request by devin-petersohn**
There were a significant number of unnecessary extra lines, formatting errors/inconsistencies, etc. 

The content of the tests did not change (with the exception of `quantile` getting additional percentiles to test). Rather the format of the tests are now such that it the tests are in a more maintainable order.

Also for the tests only `import pandas as pd` to `import pandas` and `import ray.dataframe as rdf` to `import ray.dataframe as pd`

### CODE REVIEW

**Review by kunalgosar — APPROVED** [review]
Looks great!

**Comment by pcmoritz** [comment]
We should also consider turning on yapf for DataFrames (in a separate PR, see https://github.com/ray-project/ray/blob/master/.travis/yapf.sh#L16)

**Comment by devin-petersohn** [comment]
Jenkins, retest this please.

File: python/ray/dataframe/test/test_dataframe.py [hunk 1]
```diff
@@ -1283,37 +1277,38 @@ def test_drop():
                                 simple[['B']])
 
     # non-unique - wheee!
```
Comment (pschafhalter) [inline python/ray/dataframe/test/test_dataframe.py]: Maybe get rid of `whee!`

File: python/ray/dataframe/test/test_dataframe.py [hunk 2]
```diff
@@ -1669,64 +1640,57 @@ def test_fillna_sanity(num_partitions=2):
...
-        ray_df.fillna(method='ffill'),
-        test_data.tsframe.fillna(method='ffill')
-    )
+    ray_df = pd.DataFrame(test_data.tsframe)
+    assert ray_df_equals_pandas(ray_df.fillna(method='ffill'),
+                                test_data.tsframe.fillna(method='ffill'))
 
 
 @pytest.fixture
 def test_bfill2(num_partitions=2):
```
Comment (pschafhalter) [inline python/ray/dataframe/test/test_dataframe.py]: Remove `num_partitions=2`

File: python/ray/dataframe/test/test_dataframe.py [hunk 3]
```diff
@@ -1669,64 +1640,57 @@ def test_fillna_sanity(num_partitions=2):
...
-        ray_df.fillna(method='bfill'),
-        test_data.tsframe.fillna(method='bfill')
-    )
+    ray_df = pd.DataFrame(test_data.tsframe)
+    assert ray_df_equals_pandas(ray_df.fillna(method='bfill'),
+                                test_data.tsframe.fillna(method='bfill'))
 
 
 @pytest.fixture
 def test_fillna_inplace(num_partitions=2):
```
Comment (pschafhalter) [inline python/ray/dataframe/test/test_dataframe.py]: Remove `num_partitions=2`.

File: python/ray/dataframe/test/test_dataframe.py [hunk 4]
```diff
@@ -1738,159 +1702,129 @@ def test_fillna_inplace(num_partitions=2):
...
-    ray_df = from_pandas(df[-2:].reindex(index), num_partitions).fillna(
-        method='backfill', limit=5
-    )
+    ray_df = pd.DataFrame(df[-2:].reindex(index)).fillna(method='backfill',
+                                                         limit=5)
     assert ray_df_equals_pandas(ray_df, expected)
 
 
 @pytest.fixture
 def test_frame_pad_backfill_limit(num_partitions=2):
```
Comment (pschafhalter) [inline python/ray/dataframe/test/test_dataframe.py]: Remove `num_partitions=2`.

File: python/ray/dataframe/test/test_dataframe.py [hunk 5]
```diff
@@ -1738,159 +1702,129 @@ def test_fillna_inplace(num_partitions=2):
...
-        ray_df.fillna(method='backfill', limit=5),
-        result.fillna(method='backfill', limit=5)
-    )
+    ray_df = pd.DataFrame(result)
+    assert ray_df_equals_pandas(ray_df.fillna(method='backfill', limit=5),
+                                result.fillna(method='backfill', limit=5))
 
 
 @pytest.fixture
 def test_fillna_dtype_conversion(num_partitions=2):
```
Comment (pschafhalter) [inline python/ray/dataframe/test/test_dataframe.py]: Remove `num_partitions=2`

File: python/ray/dataframe/test/test_dataframe.py [hunk 6]
```diff
@@ -1738,159 +1702,129 @@ def test_fillna_inplace(num_partitions=2):
...
     for v in ['', 1, np.nan, 1.0]:
-        assert ray_df_equals_pandas(
-            ray_df.fillna(v),
-            df.fillna(v)
-        )
+        assert ray_df_equals_pandas(ray_df.fillna(v), df.fillna(v))
 
 
 @pytest.fixture
 def test_fillna_skip_certain_blocks(num_partitions=2):
```
Comment (pschafhalter) [inline python/ray/dataframe/test/test_dataframe.py]: Remove `num_partitions=2`

File: python/ray/dataframe/test/test_dataframe.py [hunk 7]
```diff
@@ -1738,159 +1702,129 @@ def test_fillna_inplace(num_partitions=2):
...
     # don't try to fill boolean, int blocks
 
-    df = pd.DataFrame(np.random.randn(10, 4).astype(int))
-    ray_df = from_pandas(df, num_partitions)
+    df = pandas.DataFrame(np.random.randn(10, 4).astype(int))
+    ray_df = pd.DataFrame(df)
 
     # it works!
```
Comment (pschafhalter) [inline python/ray/dataframe/test/test_dataframe.py]: Remove comment.

File: python/ray/dataframe/test/test_dataframe.py [hunk 8]
```diff
@@ -1738,159 +1702,129 @@ def test_fillna_inplace(num_partitions=2):
...
-    assert ray_df_equals_pandas(
-        ray_df.fillna(np.nan),
-        df.fillna(np.nan)
-    )
+    assert ray_df_equals_pandas(ray_df.fillna(np.nan), df.fillna(np.nan))
 
 
 @pytest.fixture
 def test_fillna_dict_series(num_partitions=2):
```
Comment (pschafhalter) [inline python/ray/dataframe/test/test_dataframe.py]: Remove `num_partitions=2`

File: python/ray/dataframe/test/test_dataframe.py [hunk 9]
```diff
@@ -1738,159 +1702,129 @@ def test_fillna_inplace(num_partitions=2):
...
     # Series treated same as dict
-    assert ray_df_equals_pandas(
-        ray_df.fillna(df.max()),
-        df.fillna(df.max())
-    )
+    assert ray_df_equals_pandas(ray_df.fillna(df.max()), df.fillna(df.max()))
 
 
 @pytest.fixture
 def test_fillna_dataframe(num_partitions=2):
```
Comment (pschafhalter) [inline python/ray/dataframe/test/test_dataframe.py]: Remove `num_partitions=2`

File: python/ray/dataframe/test/test_dataframe.py [hunk 10]
```diff
@@ -1738,159 +1702,129 @@ def test_fillna_inplace(num_partitions=2):
...
     # only those columns and indices which are shared get filled
-    assert ray_df_equals_pandas(
-        ray_df.fillna(df2),
-        df.fillna(df2)
-    )
+    assert ray_df_equals_pandas(ray_df.fillna(df2), df.fillna(df2))
 
 
 @pytest.fixture
 def test_fillna_columns(num_partitions=2):
```
Comment (pschafhalter) [inline python/ray/dataframe/test/test_dataframe.py]: Remove `num_partitions=2`

File: python/ray/dataframe/test/test_dataframe.py [hunk 11]
```diff
@@ -1918,28 +1850,22 @@ def test_fillna_col_reordering(num_partitions=2):
...
-        ray_df.fillna('?'),
-        df.fillna('?')
-    )
-
-    df = pd.DataFrame({'A': [-1, -2, np.nan],
-                       'B': [pd.Timestamp('2013-01-01'),
-                             pd.Timestamp('2013-01-02'), pd.NaT],
-                       'C': ['foo', 'bar', None],
-                       'D': ['foo2', 'bar2', None]},
+    df = pandas.DataFrame({'A': [-1, -2, np.nan],
```
Comment (pschafhalter) [inline python/ray/dataframe/test/test_dataframe.py]: Change to `df = pandas.DataFrame(...)` and `ray_df = pd.DataFrame(df)`.

File: python/ray/dataframe/test/test_dataframe.py [hunk 12]
```diff
@@ -2122,25 +2048,25 @@ def test_itertuples(ray_df, pandas_df):
 
 
 def test_join():
-    ray_df = rdf.DataFrame({"col1": [0, 1, 2, 3], "col2": [4, 5, 6, 7],
+    ray_df = pd.DataFrame({"col1": [0, 1, 2, 3], "col2": [4, 5, 6, 7],
```
Comment (pschafhalter) [inline python/ray/dataframe/test/test_dataframe.py]: See above.

File: python/ray/dataframe/test/test_dataframe.py [hunk 13]
```diff
@@ -2122,25 +2048,25 @@ def test_itertuples(ray_df, pandas_df):
...
                            "col3": [8, 9, 0, 1], "col4": [2, 4, 5, 6]})
 
-    pandas_df = pd.DataFrame({"col1": [0, 1, 2, 3], "col2": [4, 5, 6, 7],
-                              "col3": [8, 9, 0, 1], "col4": [2, 4, 5, 6]})
+    pandas_df = pandas.DataFrame({"col1": [0, 1, 2, 3], "col2": [4, 5, 6, 7],
+                                  "col3": [8, 9, 0, 1], "col4": [2, 4, 5, 6]})
 
-    ray_df2 = rdf.DataFrame({"col5": [0], "col6": [1]})
+    ray_df2 = pd.DataFrame({"col5": [0], "col6": [1]})
```
Comment (pschafhalter) [inline python/ray/dataframe/test/test_dataframe.py]: See above.

File: python/ray/dataframe/test/test_dataframe.py [hunk 14]
```diff
@@ -2229,22 +2155,22 @@ def test_melt():
...
+    assert type(ray_df.memory_usage()) is pandas.core.series.Series
     assert ray_df.memory_usage(index=True).at['Index'] is not None
     assert ray_df.memory_usage(deep=True).sum() >= \
         ray_df.memory_usage(deep=False).sum()
 
 
 def test_merge():
-    ray_df = rdf.DataFrame({"col1": [0, 1, 2, 3], "col2": [4, 5, 6, 7],
-                            "col3": [8, 9, 0, 1], "col4": [2, 4, 5, 6]})
+    ray_df = pd.DataFrame({"col1": [0, 1, 2, 3], "col2": [4, 5, 6, 7],
```
Comment (pschafhalter) [inline python/ray/dataframe/test/test_dataframe.py]: See above.

File: python/ray/dataframe/test/test_dataframe.py [hunk 15]
```diff
@@ -2229,22 +2155,22 @@ def test_melt():
...
+                           "col3": [8, 9, 0, 1], "col4": [2, 4, 5, 6]})
 
-    pandas_df = pd.DataFrame({"col1": [0, 1, 2, 3], "col2": [4, 5, 6, 7],
-                              "col3": [8, 9, 0, 1], "col4": [2, 4, 5, 6]})
+    pandas_df = pandas.DataFrame({"col1": [0, 1, 2, 3], "col2": [4, 5, 6, 7],
+                                  "col3": [8, 9, 0, 1], "col4": [2, 4, 5, 6]})
 
-    ray_df2 = rdf.DataFrame({"col1": [0, 1, 2], "col2": [1, 5, 6]})
+    ray_df2 = pd.DataFrame({"col1": [0, 1, 2], "col2": [1, 5, 6]})
```
Comment (pschafhalter) [inline python/ray/dataframe/test/test_dataframe.py]: See above.

File: python/ray/dataframe/test/test_dataframe.py [hunk 16]
```diff
@@ -3104,32 +2984,32 @@ def test_unstack():
 
 
 def test_update():
-    df = rdf.DataFrame([[1.5, np.nan, 3.],
-                        [1.5, np.nan, 3.],
-                        [1.5, np.nan, 3],
-                        [1.5, np.nan, 3]])
+    df = pd.DataFrame([[1.5, np.nan, 3.],
```
Comment (pschafhalter) [inline python/ray/dataframe/test/test_dataframe.py]: See above.

File: python/ray/dataframe/test/test_dataframe.py [hunk 17]
```diff
@@ -3418,33 +3298,34 @@ def test_iloc(ray_df, pd_df):
...
 
 
 def test_to_datetime():
-    ray_df = rdf.DataFrame({'year': [2015, 2016],
-                            'month': [2, 3],
-                            'day': [4, 5]})
-    pd_df = pd.DataFrame({'year': [2015, 2016],
-                          'month': [2, 3],
-                          'day': [4, 5]})
+    ray_df = pd.DataFrame({'year': [2015, 2016],
```
Comment (pschafhalter) [inline python/ray/dataframe/test/test_dataframe.py]: See above.

File: python/ray/dataframe/test/test_dataframe.py [hunk 18]
```diff
@@ -3418,33 +3298,34 @@ def test_iloc(ray_df, pd_df):
...
 def test_get_dummies():
-    ray_df = rdf.DataFrame({'A': ['a', 'b', 'a'],
-                            'B': ['b', 'a', 'c'],
-                            'C': [1, 2, 3]})
-    pd_df = pd.DataFrame({'A': ['a', 'b', 'a'],
-                          'B': ['b', 'a', 'c'],
-                          'C': [1, 2, 3]})
-
-    assert ray_df_equals_pandas(rdf.get_dummies(ray_df), pd.get_dummies(pd_df))
+    ray_df = pd.DataFrame({'A': ['a', 'b', 'a'],
```
Comment (pschafhalter) [inline python/ray/dataframe/test/test_dataframe.py]: See above.

**Review by pschafhalter — CHANGES_REQUESTED** [review]
Great work! Left comments on minor stylistic changes.

Overall, I think we have a bit of a problem in terms of stylistic consistency. For example, `df` can refer to Pandas or Ray dataframes. Sometimes we use `pd_df` (which is strange now that we `import ray.dataframe as pd`), other times `ray_df`, and sometimes `expected`.

Also, sometimes we use single quotes or double quotes. If not in this one, we might want to fix this in a future PR.

I also think that we might want to move all tests that accept dataframes as arguments to a specific section of the file or even another file.

File: python/ray/dataframe/test/test_dataframe.py [hunk 19]
```diff
@@ -4,177 +4,45 @@
...
-import pandas as pd
+import pandas
 import pandas.util.testing as tm
-import ray.dataframe as rdf
-from ray.dataframe.utils import (
-    from_pandas,
-    to_pandas)
-
 from pandas.tests.frame.common import TestData
+import ray.dataframe as pd
```
[the remark this answers is not in the document]
Comment (kunalgosar) [reply python/ray/dataframe/test/test_dataframe.py]: There is an open PR: https://github.com/ray-project/ray/pull/1942

File: python/ray/dataframe/test/test_dataframe.py [hunk 20]
```diff
@@ -4,177 +4,45 @@
...
 @pytest.fixture
 def create_test_dataframe():
-    df = pd.DataFrame({'col1': [0, 1, 2, 3],
-                       'col2': [4, 5, 6, 7],
-                       'col3': [8, 9, 10, 11],
-                       'col4': [12, 13, 14, 15],
-                       'col5': [0, 0, 0, 0]})
-
-    return from_pandas(df, 2)
+    return pd.DataFrame({'col1': [0, 1, 2, 3],
```
[the remark this answers is not in the document]
Comment (kunalgosar) [reply python/ray/dataframe/test/test_dataframe.py]: At some point we need to do a rewrite of the tests, this is a known issue

[the remark this answers is not in the document]
Comment (simon-mo) [reply python/ray/dataframe/test/test_dataframe.py]: We need a good design doc on testing as well (robust and future proof, should be self-contained but does not need to be as exhaustive as pandas test, a good pytest structure that enable tests to run in parallel...). Can you draft one? In the meantime, I don't think we should refactor the tests. This is just a code formatting PR after all.

File: python/ray/dataframe/test/test_dataframe.py [hunk 21]
```diff
@@ -337,13 +207,13 @@ def test_int_dataframe():
 
 def test_float_dataframe():
 
-    pandas_df = pd.DataFrame({'col1': [0.0, 1.0, 2.0, 3.0],
-                              'col2': [4.0, 5.0, 6.0, 7.0],
-                              'col3': [8.0, 9.0, 10.0, 11.0],
-                              'col4': [12.0, 13.0, 14.0, 15.0],
-                              'col5': [0.0, 0.0, 0.0, 0.0]})
+    pandas_df = pandas.DataFrame({'col1': [0.0, 1.0, 2.0, 3.0],
```
[the remark this answers is not in the document]
Comment (kunalgosar) [reply python/ray/dataframe/test/test_dataframe.py]: As part of the robustness effort, it's probably not a good idea to use the same dataframe across all the tests. Ideally, the test dataframes will be changed.

**Comment by kunalgosar** [comment]
I think it makes sense to fix the issues that this PR directly touches, given this is just a formatting fix, this probably isn't the right place to try to do a full refactor of the test code. I particularly disagree with the comments on code that this PR does not touch, they should be fixed at some point. But this is not the right place.

I'll create an issue for tracking purposes. But, at some point we should refactor all of the test code together to make it collectively more robust.

File: python/ray/dataframe/test/test_dataframe.py (hunk 20 above)
[the remark this answers is not in the document]
Comment (kunalgosar) [reply python/ray/dataframe/test/test_dataframe.py]: Made the issue at: #2134

File: python/ray/dataframe/test/test_dataframe.py [hunk 22]
```diff
@@ -4,177 +4,45 @@
...
-
-@pytest.fixture
-def test_get_dtype_counts(ray_df, pandas_df):
-    assert(ray_df.get_dtype_counts().equals(pandas_df.get_dtype_counts()))
-
-
-@pytest.fixture
-def test_get_ftype_counts(ray_df, pandas_df):
-    assert(ray_df.get_ftype_counts().equals(pandas_df.get_ftype_counts()))
+    return to_pandas(ray_df1).equals(to_pandas(ray_df2))
```
[the remark this answers is not in the document]
Comment (devin-petersohn) [reply python/ray/dataframe/test/test_dataframe.py]: `equals` doesn't work properly, we have to use this for now.

File: python/ray/dataframe/test/test_dataframe.py (hunk 16 above)
Comment (devin-petersohn) [reply python/ray/dataframe/test/test_dataframe.py]: I don't think changing this one is necessary. It's used once.

**Review by pschafhalter — APPROVED** [review]

**Pull request merged**